### PR TITLE
Provides ability to add/update links for an event

### PR DESCRIPTION
**Pull request by akeem**
# Description

Slightly out of the shape of a standard REST endpoint post to this endpoint with a collection of links will overwrite the existing ones. 

# Issues

Fixes #
Refs #2872

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by akeem** [comment]
An adjustment to the signature stuff needs to be created here, then this should be good to go

**Comment by akeem** [comment]
I think this fills the gap for now in terms of validating event ownership. The glaring issue currently is that the use of middleware is getting out of control but I'd like feedback on the other portions

File: locksmith/src/app.js [hunk 1]
```diff
@@ -44,6 +50,13 @@ app.post(
   signatureValidationMiddleware.generateProcessor(eventConfiguration)
 )
 
+app.post(
+  /^\/events\/\S+\/links$/i,
```
Comment (julien51) [inline locksmith/src/app.js]: These regexp are still scaring me!

File: locksmith/src/operations/eventOperations.ts [hunk 2]
```diff
@@ -32,11 +33,33 @@ namespace EventOperations {
...
   export const find = async (lockAddress: string): Promise<any> => {
     return Event.findOne({
+      include: [{ all: true }],
       where: {
         lockAddress: { [Op.eq]: Normalizer.ethereumAddress(lockAddress) },
       },
     })
   }
+
+  export const addLink = async (
```
Comment (julien51) [inline locksmith/src/operations/eventOperations.ts]: nit: `addLinks`

File: locksmith/src/controllers/eventController.ts [hunk 3]
```diff
@@ -46,6 +47,27 @@ namespace EventController {
...
+    let eventLinks = req.body.message.eventLinks
+
+    const databaseEvent = await eventOperation.find(lockAddress)
+
+    if (databaseEvent.owner != Normalizer.ethereumAddress(req.owner)) {
+      return res.sendStatus(401)
+    }
+
+    try {
+      await eventOperation.addLink(lockAddress, eventLinks)
```
Comment (julien51) [inline locksmith/src/controllers/eventController.ts]: nit: `addLinks`

**Pull request merged**